### DISCUSSION TITLE
Reading Additional DTSU666 in FoxEss_Modbus Integrated Environment

### DISCUSSION THREAD

**Discussion by jschmitz-de** [nathanmarlor/foxess_modbus#697] [Q&A] [answered]
Hi everyone, 
i´m glad that my H3Pro is running and all registers are available in HA,
By a stroke of luck, I have another DTSU666 that I would like to use as an intermediate meter for one floor. 
Can I use the same USR W610 to read another meter that has a different ID to the H3Pro inverter? 

tnx & kr
Joerg

**Comment by FozzieUK** [comment, marked as answer]
Hi, the USR610 isn’t really up to this, it doesn’t handle multi-access and is not quick enough to deal with the increased comms.

If you look in this integrations wiki - the Waveshare device is probably best for this, alternatively the USR TCP232 410s

**Reply by jschmitz-de** [reply]
Thanks for your feedback .. i Think i will give waveshare a try

**Comment by rand12345** [comment]
If you simply want a power reading wire the DSU666 to the RS485 port on the H3 Pro and set the address to 2, you will then see this load as CT2/Meter2 value in HA.

This hardware feature is used for monitoring a second solar inverter and does not affect the calculations or operation of the H3 AFAIK, not sure about cloud reporting though. [See page 7 of the manual for more info.](https://midsummerwholesale.co.uk/pdfs/en-h3-pro-user-manual-20240410-.pdf)

**Reply by jschmitz-de** [reply]
I think i´ll try the seperate RS485 Line with a waveshare device - perhalps i find some money and bring up a parallel Inverter in the future...
Because of that it make sense to decouple "Inverter ModBus" from Measurement toy - stuff :-)
thx
